src/readii/feature_extraction.py

```diff
@@ -190,131 +192,140 @@ def radiomicFeatureExtraction(
+                                        ctDirPath, segFilePath, ctImage, roiImage
+                                    )
+                                    logger.warning(
+                                        f"Padded segmentation to match CT for {patID}."
+                                        "roiImage.GetSize() after padding: {roiImage.GetSize()}"
+                                    )
+                                else:
+                                    raise RuntimeError(
+                                        "CT and ROI dimensions do not match."
+                                    )
+
+                        # Catching CT and segmentation size mismatch error
+                        except RuntimeError as e:
+                            logger.error(str(e))
+
+                        # Extract radiomic features from this CT/segmentation pair
+                        idFeatureVector = singleRadiomicFeatureExtraction(
+                            ctImage,
+                            roiImage=roiImage,
+                            pyradiomicsParamFilePath=pyradiomicsParamFilePath,
+                            negativeControl=negativeControl,
+                            randomSeed=randomSeed
+                        )
+
+                        # Create dictionary of image metadata to append to front of output table
+                        sampleROIData = {
+                            "patient_ID": patID,
+                            "study_description": segSeriesInfo.iloc[0][
+                                "study_description_CT"
+                            ],
+                            "series_UID": segSeriesInfo.iloc[0]["series_CT"],
+                            "series_description": segSeriesInfo.iloc[0][
+                                "series_description_CT"
+                            ],
+                            "image_modality": segSeriesInfo.iloc[0]["modality_CT"],
+                            "instances": segSeriesInfo.iloc[0]["instances_CT"],
+                            "seg_series_UID": segSeriesInfo.iloc[0]["series_seg"],
+                            "seg_modality": segSeriesInfo.iloc[0]["modality_seg"],
+                            "seg_ref_image": segSeriesInfo.iloc[0]["reference_ct_seg"],
+                            "roi": roiImageName,
+                            "roi_number": roiNum,
+                            "negative_control": negativeControl,
+                        }
+
+                        # Concatenate image metadata with PyRadiomics features
+                        sampleROIData.update(idFeatureVector)
+                        # Store this ROI's info in the segmentation level list
+                        ctAllData.append(sampleROIData)
+
+            return ctAllData
+            ###### END featureExtraction #######
+        except Exception as e:
+            if keep_running:
+                print(f"Error processing patient {patID}, series {ctSeriesID}: {e}")
```

Comment:
Need to use the logger for this - see line 124 (error in singleRadiomicFeatureExtraction)